### ISSUE TITLE
Merge 2.3 changes into EE4J_8 branch

### ISSUE THREAD

**Issue by arjantijms** [eclipse-ee4j/mojarra#4380]
A number of changes from the 2.3.3 branch have not been merged into the old master, and hence didn't turn up in the EE4J_8 branch. This should be done before any work is continued in EE4J_8 (JSF 2.3).

**Comment by ren-zhijun-oracle** [comment]
@arjantijms , when will you have time to do this?

**Comment by arjantijms** [comment]
Hi Zhijun,

I took a brief look this week. It looks like not all 109 commits seem to be
things the master is actually missing, as many of them are from before
2.3.3 and master should have been up to date with that.

It should just about be doable to check them manually one by one, but this
is also not a trivial case of running git merge. I'm planning to do this
this weekend (tomorrow and the day after tomorrow).

Kind regards,
Arjan

**Comment by BalusC** [comment]
EE4J_8 branch slipped through during the Oracle-Eclipse transfer. In the end it never made it into Mojarra 2.3. The sole changes are already done via a new issue #4802. The EE4J_8 branch can be safely deleted now.
